src/main/kotlin/no/nav/familie/ba/sak/integrasjoner/sanity/SanityService.kt

```diff
@@ -1,33 +1,66 @@
 package no.nav.familie.ba.sak.integrasjoner.sanity
 
-import no.nav.familie.ba.sak.kjerne.brev.BrevKlient
+import no.nav.familie.ba.sak.kjerne.brev.domene.ISanityBegrunnelse
 import no.nav.familie.ba.sak.kjerne.brev.domene.SanityBegrunnelse
+import no.nav.familie.ba.sak.kjerne.vedtak.begrunnelser.EØSStandardbegrunnelse
+import no.nav.familie.ba.sak.kjerne.vedtak.begrunnelser.IVedtakBegrunnelse
 import no.nav.familie.ba.sak.kjerne.vedtak.begrunnelser.SanityEØSBegrunnelse
-import no.nav.familie.ba.sak.task.OpprettTaskService.Companion.RETRY_BACKOFF_5000MS
-import org.springframework.core.env.Environment
-import org.springframework.retry.annotation.Backoff
-import org.springframework.retry.annotation.Retryable
+import no.nav.familie.ba.sak.kjerne.vedtak.begrunnelser.Standardbegrunnelse
+import org.slf4j.LoggerFactory
+import org.springframework.cache.annotation.Cacheable
 import org.springframework.stereotype.Service
 
 @Service
 class SanityService(
-    private val cachedSanityKlient: CachedSanityKlient,
-    private val brevKlient: BrevKlient,
-    private val environment: Environment,
+    private val sanityKlient: SanityKlient,
 ) {
 
-    @Retryable(
-        value = [Exception::class],
-        maxAttempts = 3,
-        backoff = Backoff(delayExpression = RETRY_BACKOFF_5000MS),
-    )
-    fun hentSanityBegrunnelser(): List<SanityBegrunnelse> = cachedSanityKlient.hentSanityBegrunnelserCached()
+    private val logger = LoggerFactory.getLogger(javaClass)
 
-    @Retryable(
-        value = [Exception::class],
-        maxAttempts = 3,
-        backoff = Backoff(delayExpression = RETRY_BACKOFF_5000MS),
-    )
-    fun hentSanityEØSBegrunnelser(): List<SanityEØSBegrunnelse> =
-        cachedSanityKlient.hentEØSBegrunnelserCached()
+    @Cacheable("sanityBegrunnelser", cacheManager = "shortCache")
+    fun hentSanityBegrunnelser(): Map<Standardbegrunnelse, SanityBegrunnelse> {
+        val enumPåApiNavn = Standardbegrunnelse.values().associateBy { it.sanityApiNavn }
+        val sanityBegrunnelser = sanityKlient.hentBegrunnelser()
+        logManglerSanityBegrunnelseForEnum(enumPåApiNavn, sanityBegrunnelser, "SanityBegrunnelse")
+        return sanityBegrunnelser
+            .mapNotNull {
+                val enum = enumPåApiNavn[it.apiNavn]
```

Comment:
Burde vi unngå å bruke variabelnavn som også er keywords i kotlin?
```suggestion
                val begrunnelseEnum = enumPåApiNavn[it.apiNavn]
```

Reply:
Enig! https://github.com/navikt/familie-ba-sak/pull/3662/commits/5deb30b45fa72fd59c0464c78eb1f3ff4a6ce23a